lib/semian/lru_hash.rb

```diff
@@ -50,7 +67,7 @@ def set(key, resource)
       @table[key] = resource
       resource.updated_at = Time.now
     end
-    clear_unused_resources
+    clear_unused_resources if @table.length > @max_size
```

Comment:
This prevents the clearing from being run unnecessary, however, this doesn't actually prevent it from growing big. If I query 100,000 resources (for the sake of example) within 300 seconds, they won't get cleared. Isn't the point here to actually limit it at the top of the set and then force removals? This may not be blocking now (since it doesn't seem to be the case), but I'd expect that invariant to be preserved.

Reply:
I called this out specifically:

```
# Note:
  #   The +min_time+ is a stronger guarantee than +max_size+. That is, if there are
  #   more than +max_size+ entries in the cache, but they've all been updated more
  #   recently than +min_time+, the garbage collection will not remove them and the
  #   cache can grow without bound. This usually means that you have many active
  #   circuits to disparate endpoints (or your circuit names are bad).
```

It's a longer conversation about what to do in this case, because it means evicting an `in_use` resource.

Reply:
I missed that, sorry. I think it's acceptable then 👍 Having used 1,000 different resources for 300 seconds is a different one every 3 seconds. That seems pretty high and very reasonable.